portable/CMakeLists.txt

```diff
@@ -983,12 +990,33 @@ target_include_directories(freertos_kernel_port PUBLIC
     $<$<STREQUAL:${FREERTOS_PORT},WIZC_PIC18>:${CMAKE_CURRENT_LIST_DIR}/WizC/PIC18>
 )
 
+if(FREERTOS_PORT STREQUAL GCC_POSIX)
+    find_package(Threads REQUIRED)
+endif()
+
 target_link_libraries(freertos_kernel_port
     PUBLIC
         $<$<STREQUAL:${FREERTOS_PORT},GCC_RP2040>:pico_base_headers>
         $<$<STREQUAL:${FREERTOS_PORT},GCC_XTENSA_ESP32>:idf::esp32>
     PRIVATE
         freertos_kernel
+        $<$<STREQUAL:${FREERTOS_PORT},GCC_POSIX>:Threads::Threads>
         "$<$<STREQUAL:${FREERTOS_PORT},GCC_RP2040>:hardware_clocks;hardware_exception>"
         $<$<STREQUAL:${FREERTOS_PORT},MSVC_MINGW>:winmm> # Windows library which implements timers
 )
+
+# Note these are primarily for the POSIX port.
```

Comment:
Same as above.

Reply:
I would rather leave this to the caller to handle. We don't want to unexpectedly mask any warnings.

Reply:
The goal here was to in at least Clang and GNU compilers - enforce ALL warnings to be defined and then identify specifically the warnings that the `FreeRTOS-Kernel` group is comfortable masking.  All of these are currently masked in the current build since -Wall and -Wpedantic are currently not used.

The goal would be in further pull requests to:
- Fix any of the identified issues, OR 
- Identify the issues that you are willing to allow and move to the top level CMakeLists.txt:
  - eg: `-Wno-padded` and `-Wno-reserved-macro-identifier`
- Mask the ones out in header files that you are comfortable with by using local comments:
  - eg: 
```c
// NOLINTNEXTLINE disable-macro-expansion
```
- Note if you want them local to a particular port then I suggest making a CMakeLists.txt per port and then aliasing the `freertos_kernel_port` target to that particular port.

Reply:
I agree that which warnings to suppress should be decided by the application writer.

Reply:
@aggarg and @paulbartell - How do you propose an application writer decide what the `FreeRTOS-Kernel` developer has deemed acceptable in the `freertos_kernel` code base? And how does the application writer decides what is an expected concern/lint warning and what is not?

As it stands now, compiling the `freertos_kernel` target has decided to ignore many warnings and linter checks through the omission of `-Wpedantic, -Wall, -Weverything` and using the default settings/warnings of the GNU and Clang compiler.
This PR is attempting to fix that issue and move to an opt-out approach instead of an opt-in approach so that the application developer isn't burdened with trying to figure out the assumptions of the `freertos_kernel` developer.

These suppressed warnings are a sub set of the warnings that were previously suppressed.

As mentioned above, the intent here is to inform what is currently necessary to suppress to allow compilation to occur with the `-Wall, -Wpedantic, etc.` enabled and without error.  At some point in the future, additional changes can be added to fix these issues and then remove the need to explicitly state that these are suppressed.

It allows further code changes to adhere to a stricter standard of coding.

If one continues to omit the `-Wall, -Weverything, -Wpedantic, etc.` the FreeRTOS developer is now dictating to the application developer that they are unable to compile the entire design with these settings for the compilers without manually adding these types of additions to each of the targets in the FreeRTOS-Kernel code base, thus causing a significant amount of work for every application developer as opposed to leaving these in and adding further ones for other compilers as they are added to the test suite of compilation (eg. CI/CD compile for ARMClang, CCS, IAR, etc.).  I only added the compilers that I have access to since the CI/CD environment does not actively test each and every compiler and port variation.

Reply:
@phelter An application writer can always apply the relevant compiler flags to a target using [target_compile_options](https://cmake.org/cmake/help/latest/command/target_compile_options.html).

This is particularly important since newer versions of gcc add new warnings (and suppression flags) all the time. So adding warning suppressions here is likely to cause build failures with older toolchains.

Reply:
Thanks @paulbartell - I've removed them and will unfortunately have to add them into every application project now consuming freertos kernel.

How does FreeRTOS-Kernel check and identify the supported compilers and versions will in-fact compile?